scripts/fixate-requirements.py

```diff
@@ -138,19 +144,34 @@ def merge_source_requirements(sources):
     merged_requirements = []
     for infile_path in (locate_file(p, must_exist=True) for p in sources):
         for req in load_requirements(infile_path):
-            # Requirements starting with project name "project ..."
-            if req.req:
-                # Skip already added project name
-                if req.name in projects:
-                    continue
-                projects.add(req.name)
-                merged_requirements.append(req)
-
-            # Requirements lines like "vcs+proto://url"
-            elif req.link:
-                merged_requirements.append(req)
+            if hasattr(req, "requirement"):
+                # Requirements starting with project name "project ..."
+                parsedreq = parse_req_from_line(req.requirement, req.line_source)
+                if parsedreq.requirement:
+                    # Skip already added project name
+                    if parsedreq.requirement.name in projects:
+                        continue
+                    projects.add(parsedreq.requirement.name)
+                    merged_requirements.append(req)
+
+                    # Requirements lines like "vcs+proto://url"
+                elif parsedreq.link:
```

Comment:
Is black indenting this comment? I had to reread it a couple times to realize it was for the next line, not the current block.
```suggestion
                # Requirements lines like "vcs+proto://url"
                elif parsedreq.link:
```